server/src/main/java/fr/uga/l3miage/tp4/repositories/CandidateRepository.java

```diff
@@ -0,0 +1,24 @@
+package fr.uga.l3miage.tp4.repositories;
+
+import fr.uga.l3miage.tp4.enums.TestCenterCode;
+import fr.uga.l3miage.tp4.models.CandidateEntity;
+import fr.uga.l3miage.tp4.models.TestCenterEntity;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.Set;
+
+@Repository
+public interface CandidateRepository extends JpaRepository<CandidateEntity, Long> {
+
+    Set<CandidateEntity> findAllByTestCenterEntity_Code(TestCenterCode code) ;
+
+    //Set<CandidateEntity> findByCandidateEvaluationGridEntitiesGradeLessThanEqual(Double grade);
+
+    Set<CandidateEntity> findByCandidateEvaluationGridEntitiesGradeLessThanEqual(Double grade);
```

Comment:
Effectivement ça marche, mais ton set aura toujours qu'un résultat, il fallait faire un `findAll`
```suggestion
Set<CandidateEntity> findAllByCandidateEvaluationGridEntitiesGradeLessThanEqual(Double grade);
```